src/components/ToggleSwitch.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+
+const bgLight = `bg-[url('./assets/light.svg')]`
+const bgDark = `bg-[url('./assets/dark.svg')]`
+
+function ToggleSwitch({ isToggled, onChange }) {
+  const [isOn, setIsOn] = useState(isToggled)
```

Comment:
I suggest to control Toggle position outside of component.
Toggle can be "dummy" component, and receive its position via props.
Parent component can store state for this Toggle.